### PR TITLE
feat(components): add `underline` prop to Text component

### PR DESCRIPTION
**Pull request by zbaylin**
This is another fairly simple addition. We get the underline info from Skia via FontMetrics and draw a rectangle from it.

Here's what the text looks like all underlined (before I added the property):
Screen Shot 2020-06-08 at 8 53 54 AM

I also added it to the components that use `Text`, like `ClickableText` and `Link`. I made `Link`s underlined by default and `ClickableText`s not.

### CODE REVIEW

File: src/UI/TextNode.re [hunk 1]
```diff
@@ -180,22 +195,29 @@ class textNode (text: string) = {
...
       | {width: textWidth, _} as style
           when textWidth == Layout.Encoding.cssUndefined =>
         _this#handleTextWrapping(width, style)
       | {textOverflow: Ellipsis | UserDefined(_), _} =>
         _this#textOverflow(float_of_int(width))
       | style => _this#handleTextWrapping(width, style)
-      }
-    );
+      };
+    _textWidth = dims.width;
```
Comment (bryphe) [inline src/UI/TextNode.re:219]: Is this related to the underline? Or does this fix an unrelated bug?

File: src/UI_Primitives/Text.re [hunk 2]
```diff
@@ -20,6 +20,7 @@ let%nativeComponent make =
                       ~italic=false,
                       ~monospaced=false,
                       ~fontSize=14.,
+                      ~underlined=false,
```
Comment (bryphe) [inline src/UI_Primitives/Text.re:23]: Nice! 🎉

File: src/UI/TextNode.re (hunk 1 above)
Comment (bryphe) [reply src/UI/TextNode.re:219]: Ah, I think I understand - this is saving the width so we know how to draw the underline

How does the underline worked in the wrapped case? Is it possible for the wrapped lines to be different lengths?

File: src/Font/FontMetrics.re [hunk 3]
```diff
@@ -3,17 +3,30 @@ type t = {
...
+  underlinePos: float,
+  underlineThickness: float,
 };
 
 let empty = (size: float) => {
   height: size,
   lineHeight: size,
   ascent: 0.,
   descent: 0.,
+  underlinePos: 0.,
```
Comment (glennsl) [inline src/Font/FontMetrics.re:15]: Nitpicky, but I would prefer this to be spelled out, as it doesn't seem like this is going to be used all that frequently.

File: src/UI/TextNode.re [hunk 4]
```diff
@@ -180,22 +195,29 @@ class textNode (text: string) = {
...
+    _underlined = underlined;
+  };
+  pub measure = (width, _height): LayoutTypes.dimensions => {
     _isMeasured = true;
     /**
          If the width value is set to cssUndefined i.e. the user did not
          set a width then do not attempt to use textOverflow
        */
-    (
+    let dims =
```
Comment (glennsl) [inline src/UI/TextNode.re:210]: Also this:

```suggestion
    let dimensions =
```

I find type annotations to be much less useful and necessary when variables are properly named.

File: src/UI/TextNode.re [hunk 5]
```diff
@@ -180,22 +195,29 @@ class textNode (text: string) = {
...
           when textWidth == Layout.Encoding.cssUndefined =>
         _this#handleTextWrapping(width, style)
       | {textOverflow: Ellipsis | UserDefined(_), _} =>
         _this#textOverflow(float_of_int(width))
       | style => _this#handleTextWrapping(width, style)
-      }
-    );
+      };
+    _textWidth = dims.width;
+    dims;
```
Comment (glennsl) [inline src/UI/TextNode.re:220]:
```suggestion
    dimensions;
```

File: src/UI/TextNode.re (hunk 1 above)
Comment (glennsl) [inline src/UI/TextNode.re:219]:
```suggestion
    _textWidth = dimensions.width;
```

Comment (zbaylin) [reply src/UI/TextNode.re:219]: Yeah it definitely is, thanks for pointing that out. I'll run the text measurement on each line

Comment (zbaylin) [reply src/UI/TextNode.re]: I fixed this in the latest commits!

File: src/Font/FontMetrics.re (hunk 3 above)
Comment (zbaylin) [reply src/Font/FontMetrics.re]: Sounds good. I changed it in the latest commits (along with all the other naming suggestions)
